python/desc/imsim/skyModel.py

```diff
@@ -96,15 +98,16 @@ def addNoiseAndBackground(self, image, bandpass=None, m5=None,
 
         bandPassName = self.obs_metadata.bandpass
         skyMagnitude = skyModel.returnMags()[bandPassName]
+        skySpec = skyModel.returnWaveSpec()
 
         # Since we are only producing one eimage, account for cases
         # where nsnap > 1 with an effective exposure time for the
         # visit as a whole.  TODO: Undo this change when we write
         # separate images per exposure.
-        exposureTime = photParams.nexp*photParams.exptime*u.s
+        exposureTime = photParams.nexp*photParams.exptime
 
-        skyCounts = skyCountsPerSec(surface_brightness=skyMagnitude,
-                                    filter_band=bandPassName)*exposureTime
+        # bandpass is the CatSim bandpass object
+        skyCounts = skyCountsPerSec(skySpec, bandpass, photParams)*exposureTime*u.s
```

Comment:
You are passing `bandpass` here, but the default value of `bandpass` is `None` in the `addNoiseAndBackground` function signature.  However, `.calcADU` expects a `Bandpass` object and crashes if `None` is passed, so `bandpass` either as to be changed to be a required argument (i.e., not a keyword argument) or if it is left as `None`, then an acceptable value of `bandpass` needs to be set and passed to `calcADU`.  More generally though, we need to think somewhat more carefully about where the appropriate `Bandpass` object(s) should be created.  Currently, the bandpass is only expressed as a string, i.e., `'u', 'g', 'r', 'i', 'z', or 'y'`, in the calling code.

Reply:
OK I was going to create a bandpass dictionary in the sky counts function using photutils' galBandpassDict function and then grab the right one with the string name still as an argument of the skyCountsPerSec function, or is it better to create the dictionary outside the skyCountsPerSec function ? thank you !

Reply:
The risk with instantiating the dict in the `skyCountsPerSec` function is that, each time that function is called, you will be doing I/O reading in the dict.  Fortunately, in python, functions can have attributes (that may not be what they're called, but they function pretty similarly).  This will allow you to only instantiate the dict once using code like

```
def skyCountsPerSec(*args, **kwargs):

    if not hasattr(skyCountsPerSec, '_bp_dict'):
        skyCountsPerSec._bp_dict = [code to create the dict]
```
you can then refer to the dict as `skyCountsPerSec._bp_dict` within `skyCountsPerSec` and have access to it.